src/core/tsi/ssl_transport_security.cc

```diff
@@ -189,6 +192,11 @@ static void verified_root_cert_free(void* /*parent*/, void* ptr,
 static void init_openssl(void) {
 #if OPENSSL_VERSION_NUMBER >= 0x10100000
   OPENSSL_init_ssl(0, nullptr);
+  // Ensure OPENSSL global clean up happens after gRPC shutdown completes.
+  // OPENSSL registers an exit handler to clean up global objects, which
+  // otherwise may happen before gRPC removes all references to OPENSSL. Below
+  // exit handler is guaranteed to run after OPENSSL's.
+  std::atexit([]() { grpc_wait_for_shutdown_with_timeout(absl::Seconds(2)); });
```

Comment:
Do we want 2 seconds be hardcoded or exposed via some mechanism (api / flag)?

Reply:
I think we don't need to yet. For now I think we want to keep this workaround internal and it's only used on one place here.